osdk/tools/docker/gen_dockerfile.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: MPL-2.0
+
+# The Dockerfile generator for OSDK.
+
+output_dir="ubuntu"
+base_image="ubuntu:22.04"
+install_qemu_ovmf=true
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --help)
+            echo "Usage: gen_dockerfile.sh [OPTIONS]"
+            echo ""
+            echo "Options:"
+            echo "  --help          Print the help information"
+            echo "  --intel-tdx     Include Intel TDX support"
+            echo "  --out-dir       Output the Dockerfile under this directory. By default, the"
+            echo "                  output directory is the ubuntu directory."
+            exit 0
+            ;;
+        --intel-tdx)
+            base_image="intelcczoo/tdvm:ubuntu22.04-mvp_2023ww15"
+            install_qemu_ovmf=false
+            ;;
+        --out-dir)
+            shift
+            output_dir=$1
+            ;;
+        *)
+            echo "Unknown option: $1"
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+qemu_ovmf_installation="ovmf \\ 
+    qemu-system-x86 \\ 
+    "
+
+if [ "$install_qemu_ovmf" = false ]; then
+    qemu_ovmf_installation=""
+fi
+
+docker_file=$(cat <<EOF
+# This image is for the OSDK GitHub CI. 
+# The environment is consistent with the one 
+# described in the OSDK User Guide section of the Asterinas Book.
+#
+# TODO: We should build the Asterinas image based on the OSDK image
+# since Asterinas is managed by OSDK itself.
+# However, currently, these two images have different contents.
+# The main distinction is that
+# QEMU, grub, and OVMF in the OSDK image are installed via apt,
+# while these tools in the Asterinas image are built from source.
+# Some boot methods in Asterinas only function properly
+# when using the tools that are built from source.
```

Comment:
Please also move these comments into the `gen_dockerfile.sh` script, not the generated file.